backend/src/user.ts

```diff
@@ -0,0 +1,41 @@
+import { gen } from './code';
+import { Request, Response } from 'express';
+
+export const games: Map<string, Map<string, string>> = new Map();
+
+export function userHandle(gameId: string, req: Request, res: Response) {
+	const { body } = req;
+
+	if (!body.username) {
+		res.status(400).send();
+		return;
+	}
+
+	if (!games.has(gameId)) {
+		res.status(404).send();
+		return;
+	}
+
+	let playerSet = games.get(gameId);
+
+	if (playerSet === undefined) {
+		playerSet = new Map();
+		games.set(gameId, playerSet);
+	}
+
+	const username = body.username;
+	if (playerSet.has(username)) {
+		res.status(409).send();
```

Comment:
We should have human-readable messages here but I underspecified that in the schema. I'll think about how that should look and PR a new version later.

Reply:
Agreed. I can wait to merge to remove and add proper comments